config_prod.go

```diff
@@ -46,18 +47,37 @@ func NewStores(cfg *Config, clock clock.Clock) (*stores, error) {
 			err)
 	}
 
+	firewallDB, err := firewalldb.NewBoltDB(
+		networkDir, firewalldb.DBFilename, sessStore,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating firewall DB: %v", err)
+	}
```

Comment:
Hmmmm this is pre-existing, so I'll let you decide if you'd want to address this or not:
One issue that I realized we have here, is that if this `firewalldb.NewBoltDB` call, or the  `session.NewDB` function call above returns an `err`, we will never `Close` the previously initialized `dbs`. That is because we'll return `nil` for the `*stores` return param, and therefore not run the `close` function in `terminal.go`.

Not sure if that's something we'd want to address or not, so will let you decide on that :)!

Reply:
great catch! yes this absolutely should be addressed. But I will do it in a follow up since these commits have already been approved and since the issue was introduced before (my bad!).